menuinst/platforms/win.py

```diff
@@ -293,6 +335,45 @@ def _process_command(self, with_arg1=False) -> Tuple[str]:
             command.append("%1")
         return WinLex.quote_args(command)
 
+    def _add_remove_windows_terminal_profile(self, location: Path, remove: bool = False):
+        """Add/remove the Windows Terminal profile.
+
+        Windows Terminal is using the name of the profile to create a GUID,
+        so the name will be used as the unique identifier to find existing profiles.
+        """
+        if not self.metadata.get("terminal_profile") or not location.exists():
+            return
+        name = self.render_key("terminal_profile")
+
+        settings = json.loads(location.read_text())
+
+        index = -1
+        for p, profile in enumerate(settings["profiles"]["list"]):
+            if profile.get("name") == name:
+                index = p
+                break
+
+        if remove:
+            if index < 0:
+                log.warn(f"Could not find terminal profile for {name}.")
+                return
+            del settings["profiles"]["list"][index]
+        else:
+            profile_data = {
+                "commandline": " ".join(WinLex.quote_args(self.render_key("command"))),
+                "name": name,
+            }
+            if self.metadata.get("icon"):
+                profile_data["icon"] = self.render_key("icon")
+            if self.metadata.get("working_dir"):
+                profile_data["startingDirectory"] = self.render_key("working_dir")
+            if index < 0:
+                settings["profiles"]["list"].append(profile_data)
+            else:
+                log.warn(f"Overwriting terminal profile for {name}.")
+                settings["profiles"]["list"][index] = profile_data
+        location.write_text(json.dumps(settings, indent=4))
```

Comment:
What's the default indentation level in the factory JSON? I would have expected `2`.

```suggestion
        location.write_text(json.dumps(settings, indent=2))
```

Reply:
It is four spaces without EOL at the end of the file.